packages/holocron/src/holocronModule.jsx

```diff
@@ -17,27 +17,88 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
 import hoistStatics from 'hoist-non-react-statics';
-import { INIT_MODULE_STATE } from './ducks/load';
 
 import {
   LOAD_KEY,
   REDUCER_KEY,
   MODULES_STORE_KEY,
-} from './constants';
+  INIT_MODULE_STATE,
+} from './ducks/constants';
+
+// Execute deprecated load function and provide deprecation message
+export function executeLoad({ load, ...restProps } = { }) {
+  if (load) {
+    console.warn('The \'load\' function in holocron has been deprecated. Please use \'loadModuleData\' instead.');
+    return load(restProps);
+  }
+  return undefined;
+}
+
+// Dispatch loadModuleData inside a thunk if it exists
+export function executeLoadModuleData(loadModuleData, WrappedComponent, props) {
+  const { dispatch, ...restProps } = props;
+  if (dispatch && loadModuleData) {
+    return dispatch(async (_, getState, { fetchClient }) => loadModuleData({
+      store: { dispatch, getState },
+      fetchClient,
+      ownProps: restProps,
+      module: WrappedComponent,
+    }));
+  }
+  return undefined;
+}
+
+export function getName(WrappedComponent, name) {
+  return WrappedComponent && (WrappedComponent.displayName || WrappedComponent.name || name);
+}
 
+export function getDisplayName(name) {
+  return `HolocronModule(${name})`;
+}
+
+export async function executeLoadingFunctions({
+  // Provide loadModuleData function
+  loadModuleData,
+  // Provide WrappedComponent for loadModuleData
+  WrappedComponent,
+  // Frozen props as of when called
+  frozenProps,
+  // Provide loadCount to limit state changes
+  loadCount,
+  // Provide componentName for error messages
+  componentName,
+  // Provide component instance for getting current state of component post async
+  hocInstance,
+}) {
+  try {
+    // Run both operations in parallel if one or both exist
+    await Promise.all([
+      // Call deprecated load function
+      executeLoad(frozenProps),
+      // Call loadModuleData
+      executeLoadModuleData(loadModuleData, WrappedComponent, frozenProps),
```

Comment:
In theory if `load` were to take longer to resolve than `executeLoadModuleData` this could be a drawback. My thinking is that we should resolve both data loading functions in parallel before updating the Module.

Reply:
both could cause in conflicting updates to props and result in unexpected props

Reply:
How so? `load` is a side-effect and so is `loadModuleData`. Do either of these functions modify props? Not last I checked.

Reply:
I call `loadModuleData` and `load` serially now in 321edf9 for some predictability.